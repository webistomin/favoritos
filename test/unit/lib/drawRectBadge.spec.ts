import Favoritos from 'src/favoritos';

describe('Favoritos: drawRectBadge method', () => {
  beforeEach(() => {
    document.head.innerHTML = `<link rel="shortcut icon" href="#">`;
  });

  it('Draw rect badge if params are correct', () => {
    const lib = new Favoritos();
    lib['badgeContent'] = 1;
    expect(() => lib['drawRectBadge'](10, 22)).not.toThrow(TypeError);
    expect(() => lib['drawRectBadge'](10, 22)).not.toThrow(DOMException);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });

  it('Calculate final badge width with huge text width correctly', () => {
    const lib = new Favoritos();
    lib['badgeContent'] = 1;
    lib['options']['icon']['width'] = 32;
    expect(() => lib['drawRectBadge'](64, 22)).not.toThrow(TypeError);
    expect(() => lib['drawRectBadge'](64, 22)).not.toThrow(DOMException);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });

  it('Calculate final badge width if text width more than badge min width correctly', () => {
    const lib = new Favoritos();
    lib['badgeContent'] = 1;
    lib['options']['badge']['minWidth'] = 22;
    expect(() => lib['drawRectBadge'](23, 22)).not.toThrow(TypeError);
    expect(() => lib['drawRectBadge'](23, 22)).not.toThrow(DOMException);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });

  it('calculate final badge height correctly', () => {
    const lib = new Favoritos();
    lib['badgeContent'] = 1;
    lib['options']['badge']['minHeight'] = 22;
    expect(() => lib['drawRectBadge'](64, 64)).not.toThrow(TypeError);
    expect(() => lib['drawRectBadge'](64, 64)).not.toThrow(DOMException);
    // @ts-ignore
    expect(lib['iconCanvasContext'].__getEvents()).toMatchSnapshot();
  });
});
