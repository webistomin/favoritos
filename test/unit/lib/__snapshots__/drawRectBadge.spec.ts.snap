// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favoritos: drawRectBadge method Calculate final badge width if text width more than badge min width correctly 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 23,
      "x": 20.5,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 23,
      "x": 20.5,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
]
`;

exports[`Favoritos: drawRectBadge method Calculate final badge width with huge text width correctly 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 32,
      "x": 16,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 32,
      "x": 16,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
]
`;

exports[`Favoritos: drawRectBadge method Draw rect badge if params are correct 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 22,
      "x": 21,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
  Object {
    "props": Object {
      "height": 22,
      "width": 22,
      "x": 21,
      "y": 21,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
]
`;

exports[`Favoritos: drawRectBadge method calculate final badge height correctly 1`] = `
Array [
  Object {
    "props": Object {
      "value": "18px Helvetica, Arial, sans-serif",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "font",
  },
  Object {
    "props": Object {
      "value": "center",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textAlign",
  },
  Object {
    "props": Object {
      "value": "middle",
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "textBaseline",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 1,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "scale",
  },
  Object {
    "props": Object {
      "height": 64,
      "width": 32,
      "x": 16,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
  Object {
    "props": Object {
      "height": 64,
      "width": 32,
      "x": 16,
      "y": 0,
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "rect",
  },
]
`;
